compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1148,6 +1148,9 @@ object RefChecks {
         def strippedResultType = Applications.stripImplicit(tp.stripPoly, wildcardOnly = true).resultType
         def firstExplicitParamTypes = Applications.stripImplicit(tp.stripPoly, wildcardOnly = true).firstParamTypes
         def hasImplicitParams = tp.stripPoly match { case mt: MethodType => mt.isImplicitMethod case _ => false }
+        def isAnyAlias =
+          val denot = tp.typeSymbol.denot
+          denot.isAliasType || denot.isOpaqueAlias
```

Comment:
I'm not sure this is right.  `type Foo = Int` is an alias type, isn't it?

But if you omit the `opaque`, the warning is correct: you _can't_ ever call the extension method as an extension method.

I guess it's okay to not warn when there should be a warning, but the tricky part here is that opaque types look different depending on context.

Since I don't understand how much work is being done here and how much is being done by the later `.isOpaqueAlias` bit, I'm not sure this under-warns, but it seems like it might.

Reply:
Where the "receiver" is an alias, https://github.com/scala/scala3/blob/main/tests/warn/i16743.scala#L56

The current diff is at L1165 where the params are checked; the `frozen_<:<` is not sufficient.

For non-opaque aliases, I think the screws could be tightened where there can be no override (of the alias member).

The goal is indeed to prefer underwarn to overwarn.